### ISSUE TITLE
_X_AMZN_TRACE_ID is not set to the correct value

### ISSUE THREAD

**Issue by hamitb** [aws/aws-lambda-go#236]
Other language runtimes in Lambda provide X-Ray Trace information through the `_X_AMZN_TRACE_ID` environment variable. Recently this trace information also started to be appended to the Cloudwatch logs for each of the invocations as in the below:

![image](https://user-images.githubusercontent.com/16473976/65144154-5536b200-da1f-11e9-91fa-b60b56168b8e.png)

Even though in the other runtimes(Python, Node.js, etc.) the trace information provided through the environment variable and through the Cloudwatch logs seems to be consistent, this is not the case for the Go.

Following lambda function can be used to reproduce the issue: 
```go
package main

import (
	"context"
	"fmt"
	"os"

	"github.com/aws/aws-lambda-go/lambda"
)

type myEvent struct {
	ID string `json:"id"`
}

func handler(ctx context.Context, event myEvent) (string, error) {
	fmt.Println(os.Getenv("_X_AMZN_TRACE_ID"))
	fmt.Println(ctx.Value("x-amzn-trace-id"))

	return "Hello ƛ!", nil
}

func main() {
	lambda.Start(handler)
}
```

When I run the above lambda function, I got the values below:
![image](https://user-images.githubusercontent.com/16473976/65145152-857f5000-da21-11e9-9366-3e9395885696.png)

Here the values in the rectangle 2 and 3 are consistent as expected. However, the value provided by the environment variable(rectangle 1) does not seem to be correct.

**Comment by hamitb** [comment]
`x-amzn-trace-id` is put into context here. Maybe `_X_AMZN_TRACE_ID` environment variable can also be set to the same value here if it's an appropriate place to do so.
https://github.com/aws/aws-lambda-go/blob/02a2e049b01224f55250fc6738badc312874b250/lambda/function.go#L61

**Comment by bmoffatt** [comment]
released with v1.15.0
